forge/routes/auth/index.js

```diff
@@ -72,13 +72,17 @@ module.exports = fp(async function (app, opts, done) {
 
     app.decorate('verifyTokenOrSession', async function (request, reply) {
         // Order is important, other way round breaks nr-auth plugin
-        if (request.sid) {
-            await verifySession(request, reply)
-        } else if (request.headers && request.headers.authorization) {
-            await verifyToken(request, reply)
-        } else if (!request.context.config.allowAnonymous) {
+        try {
+            if (request.sid) {
+                await verifySession(request, reply)
+            } else if (request.headers && request.headers.authorization) {
+                await verifyToken(request, reply)
+            } else if (!request.context.config.allowAnonymous) {
+                reply.code(401).send({ error: 'unauthorized' })
```

Comment:
Is this line necessary since the throw would result in the same 401 response

Reply:
Good point - but you've also made me spot something else I need to check. Will pick it up in the morning.

Reply:
I've removed the throw. No need to add the overhead of throwing an error that is caught two lines later.

I wanted to double check that *not* throwing an error would properly stop the request being processed any further. I've confirmed that calling `reply.send()` in a `preHandler` means fastify considers the request handled and no further processing is done.